docs/csharp/csharp-7.md

```diff
@@ -16,64 +16,539 @@ ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
+you don't need to assign an initial value, and you can't accidentally use
+it before it is assigned.
+
+The scope of the variable declaration minimizes the chance for mis-using
+the variable as well.  The `out` variable is scoped to the enclosing block.
+By minimizing the scope, you avoid name collisions with other variables and
+you minimize the chance of misuse.
+
+The most common use for this feature will be the `TryParse` pattern. In this
+pattern, a method returns a `bool` indicating success or failure, and an
+`out` variable that provides the result, in cases when the method succeeds.
 
 ## Tuples
 
-### Named and unnamed Tuples
+The syntax used for classes and structs provides a rich syntax to explain
+your design intent. But sometimes that rich syntax requires extra
+work with minimal benefit. You may often write methods that need a simple
+structure containing more than one data element. These scenarios are
+why *Tuples* were added to C#. Tuples are lightweight data structures
+that contain multiple elements.  You can't define methods, you simply
+have public get/set properties (without any validation) for the
+data members on the Tuple.
+
+> [!NOTE]
+> Tuples were available before C# 7, but had many limitations. Most importantly,
+> the members of these tuples were named `Item1`, `Item2` and so on.
+
+You can create a tuple by assigning each member to a value:
+
+[!code-csharp[UnnamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#04_UnnamedTuple "Unnamed tuple")]
+
+That assignment creates a tuple whose members are `Item1` and `Item2`,
+following the existing `Tuple` syntax.
+You can modify that assignment to create a tuple that provides semantic
+names to each of the members of the tuple:
+
+[!code-csharp[NamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#05_NamedTuple "Named tuple")]
+
+> [!NOTE]
+> The new features in Tuples require the `System.ValueTuple` type. For Visual Studio 15
+> Preview 5 and earlier preview releases, you must add the NuGet package "System.ValueTuple",
+> available in the pre-release stream.
+
+The `namedLetters` tuple contains fields referred to as `alpha` and
+`beta`. In a tuple assignment, you can also specify the names of the fields
+on the right hand side of the assignment:
+
+[!code-csharp[ImplicitNamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#06_ImplicitNamedTuple "Implicitly named tuple")]
+
+The language allows you to specify names for the fields on both the
+left and right hand side of the assignment:
+
+[!code-csharp[NamedTupleConflict](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#07_NamedTupleConflict "Named tuple conflict")]
+
+The line above generates a warning, `CS8123`, telling you that the names on the right
+side of the assignment, `alpha` and `beta` are ignored because they conflict
+with the names on the left side, `first` and `second`.
+
+The examples above show the basic syntax to declare tuples. Tuples are
```

Comment:
These next two paragraphs are partly repetitive and somewhat disorganized. After "Tuples provide a simple syntax...", I'd show the example that returns the min/max values. Then follow with a bulleted list of the advantages (Simplicity of syntax -- and possibly compare it with the syntax of instantiating a tuple using a tuple constructor, which is a nightmare; no need to define class/struct; no need to create a new symbol; more efficient and more productive.